server/rooms.ts

```diff
@@ -1517,6 +1518,42 @@ export class GameRoom extends BasicChatRoom {
 		this.sendUser(connection, '|init|battle\n|title|' + this.title + '\n' + this.getLogForUser(user));
 		if (this.game && this.game.onConnect) this.game.onConnect(user, connection);
 	}
+	async saveReplay(forceUnlisted = false) {
+		const battle = this.battle;
+		if (!battle) return false;
```

Comment:
Throw?

Reply:
Why is battle nullable anyway?

Reply:
In GameRoom? Because some games aren't battles.

Reply:
which is why I didn't throw there, because it's not guaranteed to be a state that doesnt exist.

That being said, I think I prefer using room.battle as the check instead of just checking to see if it's a `GameRoom`